nexusutils/dataconverter/readers/em_spctrscpy/utils/em_generic_eln_io.py

```diff
@@ -131,19 +136,19 @@ def parse_user_section(self, template: dict) -> dict:
     def parse_sample_section(self, template: dict) -> dict:
         """Copy data in sample section."""
         # check if required fields exists and are valid
-        print('Parsing sample...')
+        print("Parsing sample...")
         src = "sample"
-        trg = "/ENTRY[entry]/sample/"
+        trg = "/ENTRY[entry" + str(self.entry_id) + "]/sample/"
         assert isinstance(self.yml[src], fd.FlatDict), \
-            'Required section ' + src + ' does not exist!'
+            "Required section " + src + " does not exist!"
         assert isinstance(self.yml[src + ":atom_types"], list), \
```

Comment:
You should use exceptions when there is a possible user error. Which it seems here is from the YAML supplied as an ELN. Assert wouldn't fit here.